lessons/creating-apis-with-python-and-flask.md

```diff
@@ -0,0 +1,713 @@
+---
+title: |
+    Creating Web APIs with Python and Flask
+collection: lessons
+layout: lesson
+slug: creating-apis-with-python-and-flask
+date: 2018-04-02
+authors:
+- Patrick Smyth
+reviewers:
+- Megan Kudzia
+- Ryan Deschamps
+editors:
+- Brandon Walsh
+difficulty: 2
+activity: presenting
+topics: [api, data-management]
+review-ticket:  106
+abstract: |
+  Learn how to set up a basic Application Programming Interface (API) to make your data more accessible to users. This lesson also discusses principles of API design and the benefits of APIs for digital projects.
+---
+
+{% include toc.html %}
+
+# Lesson Goals
+
+Web APIs are tools for making information and application functionality accessible over the internet. In this lesson, you will:
+
+- Learn what an API is and when you should use one.
+- Learn how to build a web API that returns data to its users.
+- Learn some principles of good API design, applying them to an API that draws book metadata from a database.
+
+# Setting Up
+
+## Prerequisites
+
+You can use the Windows, OSX, or Linux operating systems to complete this tutorial, and those few instructions that are not the same across platforms will be explicitly noted. Python 3, the Flask web framework, and a web browser are required for this tutorial, and installation instructions for all platforms are outlined below.
```

Comment:
OK I only have pedantic little things.

OS X has been frustratingly renamed to macOS. Might be worth doing a find and replace.